presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java

```diff
@@ -51,7 +51,7 @@
     private boolean optimizeMetadataQueries;
     private boolean optimizeHashGeneration = true;
     private boolean optimizeSingleDistinct = true;
-    private boolean optimizerReorderWindows = true;
+    private boolean optimizerReorderWindows = false;
```

Comment:
This fix is for `MergeWindows` but `ReorderWindows` is a separate issue. I needed to disable failing rewrite to be able to check if tests are passing with the fix for `MergeWindows`. It's a separate commit, though because it's not related to the this change (merge windows rule). However, this PR (this commit precisely: https://github.com/prestodb/presto/pull/7298/commits/1b0c94d86f8ac8c0e7e4cf485b4a556065841a3d) introduces test revealing the problem, so the automation would be failing without `reorder-windows` disabled. Reorder windows optimizer will be fixed separately, then enabled again.